server/src/services/requestCountsService.py

```diff
@@ -0,0 +1,70 @@
+from sqlalchemy import func
+from .dataService import DataService
+from .databaseOrm import Ingest as Request
+
+
+class RequestCountsService(object):
+    def __init__(self, config=None, tableName="ingest_staging_table"):
+        self.dataAccess = DataService(config, tableName)
+
+    async def get_req_counts(self,
+                             startDate=None,
+                             endDate=None,
+                             ncList=[],
+                             requestTypes=[],
+                             countFields=[]):
+        """
+        For each countField, returns the counts of each distinct value
+        in that field, given times, ncs, and request filters.
+        E.g. if countsFields is ['requesttype', 'requestsource'], returns:
+        {
+            'lastPulled': 'Timestamp',
+            'data': [
+                {
+                    'field': 'requesttype',
+                    'counts': {
+                        'Graffiti Removal': 'Int',
+                        'Bulky Items': 'Int',
+                        ...
+                    }
+                },
+                {
+                    'field': 'requestsource',
+                    'counts': {
+                        'Mobile App': 'Int',
+                        'Driver Self Report': 'Int',
+                        ...
+                    }
+                }
+            ]
+        }
+        """
+
+        # filter by date, nc, and requestType (if provided)
+        filters = [
+            Request.createddate > startDate if startDate else True,
+            Request.createddate < endDate if endDate else True,
+            Request.ncname.in_(ncList) if ncList else True,
+            Request.requesttype.in_(requestTypes) if requestTypes else True
+        ]
+
```

Comment:
Im thinking that we might want to adjust this to hit data service with countFields and filters and create an aggregateQuery method in dataservice....so functionality will look like 

select [countFields] from table where [filters]

Then do a pandas groupby(countFields).size()
Should follow something like this https://stackoverflow.com/a/38933130

Reply:
The reason behind this is to keep the database logic separate from any of the services which will make testing/validation much easier(when we get around to it)

Reply:
Ok, that makes sense, I can move the database logic into the dataservice. We might want to get away from constructing SQL queries by string concatenation though, it's kind of error-prone and hard to read. The ORM in sqlalchemy basically generates (and executes) SQL queries in a cleaner way.

Reply:
Yeah i agree, we can def tweak the existing query method to use the orm instead of creating a :hankey: query string

Reply:
I created #398 in the meantime so whoever takes it can use your code as an example

Reply:
ok cool, yeah I'll take that one. I can work on it while I'm creating the aggregationQuery in the dataservice. I like the aggregationQuery concept, could be useful for other endpoints down the line.

Reply:
Awesome, thanks!!
Im hoping that with query and aggregationQuery itll keep the backend to easy categorization of services. Either getting _a single thing_, _a lot of stuff_, or a _summary of groupings_
That way there wont be weird complex querying per endpoint

Reply:
Ok, I gave the aggregationQuery a shot, also removed SQL string concatenation and used ORM instead in the pins service and request detail service.